provisioning/device/src/Transports/Amqp/AmqpClientConnection.cs

```diff
@@ -282,47 +302,17 @@ private void OnReadHeaderComplete(TransportAsyncCallbackArgs args)
         private void CompleteOnException(TransportAsyncCallbackArgs args)
         {
             if (Logging.IsEnabled)
-            {
                 Logging.Enter(this, $"{nameof(AmqpClientConnection)}.{nameof(CompleteOnException)}");
-            }
 
             if (args.Exception != null && args.Transport != null)
             {
                 if (Logging.IsEnabled)
-                {
                     Logging.Error(this, $"{nameof(AmqpClientConnection)}.{nameof(CompleteOnException)}: Exception thrown {args.Exception.Message}");
-                }
 
                 args.Transport.SafeClose(args.Exception);
                 args.Transport = null;
                 _tcs.TrySetException(args.Exception);
             }
         }
-
-        public void Dispose()
-        {
-            Dispose(true);
-            GC.SuppressFinalize(this);
-        }
-
-        public virtual void Dispose(bool disposing)
-        {
-            if (_isDisposed)
-            {
-                return;
-            }
-
-            if (disposing)
-            {
-                // We don't know if the transport object is instantiated as a disposable or not
-                // We check and dispose if it is.
-                if (_transport is IDisposable disposable)
-                {
-                    disposable.Dispose();
```

Comment:
Are none of the transports disposable anymore?

Reply:
I moved the public methods higher in the file to be above the internal/private methods.

Reply:
Our transports inherit from our a delegating handler class, which is IDisposable. So both amqp and mqtt transports should be IDisposable.

Reply:
Oh, this is provisioning service, nevermind

Reply:
TransportBase (from amqp library) isn't IDisposable but our websocket implementation is